File: src/main/java/com/primogemstudio/advancedfmk/render/filter/Filter.java
package com.primogemstudio.advancedfmk.render.filter;

import com.mojang.blaze3d.pipeline.RenderTarget;

public interface Filter {
    RenderTarget getTarget();

    void setArg(String name, Object value);

    void render(float partialTicks);

    void enable();

    void reset();
}

File: src/main/java/com/primogemstudio/advancedfmk/render/filter/Filter.kt
package com.primogemstudio.advancedfmk.render.filter

import com.mojang.blaze3d.pipeline.RenderTarget

interface Filter {
    val target: RenderTarget?
    fun setArg(name: String?, value: Any?)
    fun render(partialTicks: Float)
    fun enable()
    fun reset()
}
